### DISCUSSION TITLE
is infisical suited for open source projects that want to share their env file with contributors?

### DISCUSSION THREAD

**Discussion by samuelstroschein** [Infisical/infisical#386] [Q&A]
See https://github.com/inlang/inlang/issues/312#issuecomment-1408576454

> hi there,
>
> i want to use doppler to streamline contributions for the inlang project.
>
> i set up gitpod and doppler to provide contributors with a one-click dev environment, see [Dashboard — Gitpod](https://gitpod.io/#https://github.com/inlang/inlang). gitpod allows the injection of env variable (the doppler token) into the dev environment but not for forks, see [Detailed dev setup guide · Issue #312 · inlang/inlang · GitHub 2](https://github.com/inlang/inlang/issues/312).
> 
> can i set an env file/token to public and directly commit it into git without doppler revoking the token?hi there,

**Comment by maidul98** [comment]
Hey Samuel, as I understand you would like your contributors to pull development secrets via Infisical without providing a .env file? If so, you can quickly achieve this by creating a project in Infisical and creating a service token for the development environment of that project. You can then commit this service token as a part of your development start command so then whenever contributors start up their project, it will pull in the latest secrets from Infisical. Currently, the service token has a lifetime of 1 year after which it will expire but I assume you need a permanent service token as once it is committed and forked, it won't really be changed.

I can quickly make a new option that allows you to choose a 'never expire' service token. I can have this done before Friday of this week, does this work for you? Also, you should join our slack for quicker replies https://join.slack.com/t/infisical-users/shared_invite/zt-1kdbk07ro-RtoyEt_9E~fyzGo_xQYP6g

**Reply by samuelstroschein** [reply]
> Hey Samuel, as I understand you would like your contributors to pull development secrets via Infisical without providing a .env file? 

Correct. 

To clarify, infisical never revokes a service token committed to a repository? 

> I can quickly make a new option that allows you to choose a 'never expire' service token. I can have this done before Friday of this week, does this work for you?

Not required. I'd rather have you invest the time in opening a PR to our repo that sets up our devcontainer to auto-download the latest env file :)

**Comment by maidul98** [comment]
We currently don't auto-revoke service tokens that are committed to a repo; however, this will be a great feature to have. I'll add it to our road map. 

As for exporting .env in your container, I'd be happy to do that for you. Please link me to your docker file and I'll add the PR to export the .env file into your container

**Reply by samuelstroschein** [reply]
We are using a docker image provided by VSCode https://github.com/inlang/inlang/blob/main/.devcontainer.json. Providing a custom image is possible but I am a bit hesitant to use infiscal because I would need to update things outside of the repo (which takes effort)

**Reply by maidul98** [reply]
I had a brief look at the repository but i'm unfamiliar with the docker related file you linked. I'm more familiar with docker files in these formats https://github.com/Infisical/infisical/blob/main/backend/Dockerfile. Were you able to join our slack? I think we can get this resolved quicker there https://infisical.com/slack

**Reply by samuelstroschein** [reply]
We might get rid of the docker containers altogether. The issue of adopting Infisical is not pressing. Thus, I will not move forward at the time :)
